src/contracts/core/DelegationManager.sol

```diff
@@ -201,19 +209,20 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
     }
 
     /**
-     * @notice Undelegates the staker from the operator who they are delegated to. Puts the staker into the "undelegation limbo" mode of the EigenPodManager
-     * and queues a withdrawal of all of the staker's shares in the StrategyManager (to the staker), if necessary.
+     * @notice Undelegates the staker from the operator who they are delegated to. Puts the staker into "undelegation limbo", if necessary.
      * @param staker The account to be undelegated.
-     * @return withdrawalRoot The root of the newly queued withdrawal, if a withdrawal was queued. Otherwise just bytes32(0).
      *
      * @dev Reverts if the `staker` is also an operator, since operators are not allowed to undelegate from themselves.
      * @dev Reverts if the caller is not the staker, nor the operator who the staker is delegated to, nor the operator's specified "delegationApprover"
      * @dev Reverts if the `staker` is already undelegated.
+     * @dev Note that if a staker enters undelegation limbo as a result of this function, then they will be unable to redelegate until after
+     * successfully calling `exitUndelegationLimbo`.
      */
     function undelegate(
         address staker
-    ) external onlyWhenNotPaused(PAUSED_UNDELEGATION) returns (bytes32 withdrawalRoot) {
+    ) external onlyWhenNotPaused(PAUSED_UNDELEGATION) {
         require(isDelegated(staker), "DelegationManager.undelegate: staker must be delegated to undelegate");
+        require(!isInUndelegationLimbo(staker), "DelegationManager.undelegate: staker is already in undelegation limbo");
```

Comment:
this check that the staker is not in undelegation limbo is technically redundant with the one above (that the staker *is* delegated), since that check passing already ensures this one will pass.
a new property of undelegation limbo is that a staker cannot simutaneously be both (a) in delegation limbo and (b) delegated to an operator. This is enforced with a check in the `_delegate` function.

TODO for myself: write a simple prover invariant showing that `isInUndelegationLimbo(staker) => !isDelegated(staker)`

Reply:
resolved in https://github.com/Layr-Labs/eigenlayer-contracts/pull/208/commits/7d681c2bacf58f224f9ad818e5964063f3d9f48f
leaving this thread open because I think the note about the new property/invariant is important